src/OpenSage.Game/Content/Translation/TranslationManager.cs

```diff
@@ -212,59 +214,54 @@ public string GetParticularString(string context, string str)
         public static void LoadGameStrings(FileSystem fileSystem, string language, SageGame game)
         {
             var path = string.Empty;
-            while (!(fileSystem is null))
+            switch (game)
             {
-                switch (game)
-                {
-                    case SageGame.CncGenerals:
-                    case SageGame.CncGeneralsZeroHour:
-                        path = $"Data/{language}/generals";
-                        break;
-                    case SageGame.Bfme:
-                        path = $"lang/{language}/lotr";
-                        break;
-                    case SageGame.Bfme2:
-                    case SageGame.Bfme2Rotwk:
-                        if(language=="German")
-                        {
-                            path = "lotr";
-                        }
-                        else
-                        {
-                            path = "data/lotr";
-                        }
-                        break;
-                    case SageGame.Cnc3:
-                    case SageGame.Cnc3KanesWrath:
-                        path = "cnc3";
-                        break;
-                    case SageGame.Ra3:
-                    case SageGame.Ra3Uprising: // there is a data/gamestrings_temp.csf in Uprising
-                    case SageGame.Cnc4:
-                        path = "data/gamestrings";
-                        break;
-                }
-
-                FileSystemEntry file;
-                if (!((file = fileSystem.GetFile($"{path}.csf")) is null))
-                {
-                    using var stream = file.Open();
-                    Instance.SetCultureFromLanguage(language);
-                    Instance.RegisterProvider(new CsfTranslationProvider(stream, game));
+                case SageGame.CncGenerals:
+                case SageGame.CncGeneralsZeroHour:
+                    path = Path.Combine("Data", language, "generals");
+                    break;
+                case SageGame.Bfme:
+                    path = Path.Combine("lang", language, "lotr");
+                    break;
+                case SageGame.Bfme2:
+                case SageGame.Bfme2Rotwk:
+                    if(language=="German")
+                    {
+                        path = "lotr";
+                    }
+                    else
+                    {
+                        path = Path.Combine("data", "lotr");
+                    }
+                    break;
+                case SageGame.Cnc3:
+                case SageGame.Cnc3KanesWrath:
+                    path = "cnc3";
+                    break;
+                case SageGame.Ra3:
+                case SageGame.Ra3Uprising: // there is a data/gamestrings_temp.csf in Uprising
+                case SageGame.Cnc4:
+                    path = Path.Combine("data", "gamestrings");
+                    break;
+            }
```

Comment:
May I suggest moving that into the game's definition?

Reply:
Good idea - [done](https://github.com/OpenSAGE/OpenSAGE/pull/578/commits/60b22023b18eb999952b5bbde356c88f6250d5ee).